### PR TITLE
gets the list of flavours from openstack API

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: this PR adds ability to get the list of sizes/flavours from openstack API. The route of the new endpoint is `/api/v1/openstack/sizes`. When calling this endpoint the following parameters are required `Username, Password, Tenant, Domain, DatacenterName`

### CODE REVIEW

File: api/pkg/provider/cloud/openstack/helper.go [hunk 1]
```diff
@@ -288,3 +290,27 @@ func detachSubnetFromRouter(netClient *gophercloud.ServiceClient, subnetID, rout
 	}
 	return res.Extract()
 }
+
+func getFlavor(authClient *gophercloud.ProviderClient, region, name string) ([]osflavors.Flavor, error) {
```
Comment (p0lyn0mial) [inline api/pkg/provider/cloud/openstack/helper.go]: I think that is the correct endpoint - Haven't tested this code yet. 

Oh, I almost forgot to mention that I copied this code from machine-controller 🥇

**Comment by p0lyn0mial** [comment]
FYI: I'll add more code that will consume `getFlavor` method.

File: api/pkg/provider/cloud/openstack/helper.go (hunk 1 above)
Comment (alvaroaleman) [reply api/pkg/provider/cloud/openstack/helper.go]: I'd call that `getFlavors` (with an s) unless its about getting a flavor by name/id

**Comment by p0lyn0mial** [comment]
@mrIncompetent this PR aggregates all the changes.

**Review by mrIncompetent — DISMISSED** [review]

**Comment by p0lyn0mial** [comment]
@mrIncompetent dep update.

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**